<p>Calling&nbsp;"<code>org.junit.Assert.fail()</code>" (JUnit 4) or "<code>org.junit.jupiter.api.Assertions.fail()</code>" (JUnit 5)&nbsp;throws a
"<code>java.lang.AssertionError</code>". If this call is done within the try block of a try-catch cathing a similar error, you should make sure to
test some properties of the exception. Otherwise, the test will never fail.</p>
<h2>Noncompliant Code Example</h2>
<pre>
@Test
public void should_throw_assertion_error() {
  try {
    throwAssertionError(); // This test will pass even if we comment this line!
    Assert.fail("Expected an AssertionError!");
  } catch (AssertionError e) {}
}

private void throwAssertionError() {
  throw new AssertionError("My assertion error");
}
</pre>
<h2>Compliant Solution</h2>
<pre>
assertThrows(AssertionError.class, () -&gt; throwAssertionError());
</pre>
<pre>
try {
   throwAssertionError();
   Assert.fail("Expected an AssertionError!");
 } catch (AssertionError e) {
   Assert.assertThat(e.getMessage(), is("My assertion error"));
 }
</pre>
<h2>See</h2>
<ul>
  <li> <a href="https://github.com/junit-team/junit4/wiki/Exception-testing">JUnit 4 exception testing documentation</a> </li>
</ul>

